hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -325,4 +343,69 @@ public static long instantTimeDiffSeconds(String newInstantTime, String oldInsta
     }
   }
 
+  public static DataStreamSink<?> createHoodieDataStreamSink(RowType rowType, Configuration conf, int parallelism,
+                                                             StreamWriteOperatorFactory<HoodieRecord> operatorFactory, DataStream<RowData> dataStream) {
+    DataStream<HoodieRecord> dataStream2 = dataStream.map(new RowDataToHoodieFunction<>(rowType, conf),
+            TypeInformation.of(HoodieRecord.class));
+
+    if (conf.getBoolean(FlinkOptions.INDEX_BOOTSTRAP_ENABLED)) {
+      dataStream2 = getIndexBootstrapStream(conf, parallelism, dataStream2);
```

Comment:
Move these methods into a separate class `HoodieWritePipelines` and maybe you can create a fluent API to build all kinds of sub-pipelines.